lib/dgram.js

```diff
@@ -125,6 +126,20 @@ function Socket(type, listener) {
     recvBufferSize,
     sendBufferSize
   };
+
+  if (options?.signal !== undefined) {
+    const { signal } = options;
+    validateAbortSignal(signal, 'options.signal');
+    const onAborted = () => {
+      this.close();
+    };
+    if (signal.aborted) {
+      onAborted();
+    } else {
+      signal.addEventListener('abort', onAborted);
```

Comment:
```suggestion
      signal.addEventListener('abort', onAborted, { once: true });
```

Reply:
I think this is fine, once just creates the impression that it removes the abort listener on `'abort'` but you'd need the `removeEventListener` anyway for the "server closed for another reason" case.